algorithms/nautilus/fuzzer/queue.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+/**
+ * @file queue.cpp
+ * @brief Corpus queue of Nautilus.
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
```

Comment:
I think this (original) implementation of seed queue has a lot of room for further optimizations. 
But I don't know if this implementation is being a bottleneck of this fuzzer and if we can obtain the performance that is worth the cost of re-implementation.
So leave a comment here would be enough.

Reply:
Added to TODO in [84b24a8](https://github.com/fuzzuf/fuzzuf/pull/23/commits/84b24a866646b06e63c9910a7e1225cdf6523fe6)